src/sign-typed-data.ts

```diff
@@ -168,15 +170,23 @@ function encodeField(
   }
 
   if (type === 'bytes') {
-    return ['bytes32', keccak(value)];
+    if (typeof value === 'number') {
+      value = numberToBuffer(value);
+    } else if (isHexString(value) && parseInt(value, 16) <= 16) {
```

Comment:
why is this value limited to 16 only for hex strings

Reply:
I think this was an intermediate step that we can remove now...

Reply:
removed here: https://github.com/MetaMask/eth-sig-util/pull/260/commits/ba99a2c2eedbb5f3ec8d6adae387b54eca1272d1